File: mobile/lib/model/tagihan_model.dart
class TagihanModel {
  final String kode;
  final DateTime tanggalTerbuat;
  final DateTime? tanggalBayar;
  final bool isPaid;
  final int jumlahTagihan;
  // final Appointment appointment;

  TagihanModel (
      {required this.kode,
        required this.tanggalTerbuat,
        required this.tanggalBayar,
        required this.isPaid,
        required this.jumlahTagihan,
        // required this.appointment
      });

  factory TagihanModel.fromJson(Map<String, dynamic> json) {
    return TagihanModel(
        kode: json['kode'],
        tanggalTerbuat: DateTime.parse(json['tanggalTerbuat']),
        tanggalBayar: json['tanggalBayar'] != null ? DateTime.parse(json['tanggalBayar']) : null,
        isPaid: json['isPaid'],
        jumlahTagihan: json['jumlahTagihan'],
        // appointment: json['appointment']
    );
  }

}

File: rumahsehat/src/main/java/apap/tugaskelompok/rumahsehat/model/TagihanModel.java
package apap.tugaskelompok.rumahsehat.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "tagihan")
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(value ={"appointment"}, allowSetters = true)
public class TagihanModel implements Serializable {

    @Id
    @NotNull
    private String kode;

    @NotNull
    @Column(name = "tanggal_terbuat")
    @DateTimeFormat(pattern = "dd-MM-yyyy")
    private LocalDate tanggalTerbuat;

    @Column(name = "tanggal_bayar")
    @DateTimeFormat(pattern = "dd-MM-yyyy")
    private LocalDate tanggalBayar;

    @NotNull
    @Column(name = "is_paid")
    private Boolean isPaid;

    @NotNull
    @Column(name = "jumlah_tagihan")
    private Integer jumlahTagihan;

    @NotNull
    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "kode_appointment", referencedColumnName = "kode")
    private AppointmentModel appointment;
}
